pkg/reconciler/utils/pubsub/subscription.go

```diff
@@ -71,24 +71,27 @@ func (r *Reconciler) ReconcileSubscription(ctx context.Context, id string, subCo
 	return r.createSubscription(ctx, id, subConfig, obj, updater)
 }
 
-func (r *Reconciler) DeleteSubscription(ctx context.Context, id string, obj runtime.Object) error {
+func (r *Reconciler) DeleteSubscription(ctx context.Context, id string, obj runtime.Object, updater StatusUpdater) error {
 	logger := logging.FromContext(ctx)
 	logger.Debug("Deleting decoupling sub")
 
 	sub := r.client.Subscription(id)
 	exists, err := sub.Exists(ctx)
 	if err != nil {
 		logger.Error("Failed to verify Pub/Sub subscription exists", zap.Error(err))
+		updater.MarkSubscriptionUnknown("FinalizeSubscriptionUnknown", "failed to verify Pub/Sub subscription exists: %w", err)
 		return err
 	}
 	if exists {
-		return r.deleteSubscription(ctx, sub, obj)
+		if err = r.deleteSubscription(ctx, sub, obj, updater); err != nil {
+			updater.MarkSubscriptionFailed("FinalizeSubscriptionFailed", "failed to delete Pub/Sub subscription: %w", err)
```

Comment:
In my opinion, we shouldn't mark Broker as not ready because the pull subscription still exists and the Broker works properly. Instead, once we successfully delete the pull subscription, we should MarkSubscriptionFailed.

I know this depends on how do you look at it. If you treat "BrokerBeingDeleted" in combination with "FinalizeSubscriptionFailed" as a condition, then it makes sense. However, if I am a user, I probably just look at the overall status.